tests/bluetooth/host/keys/mocks/keys_help_utils.c

```diff
@@ -17,6 +17,24 @@ void clear_key_pool(void)
 	memset(key_pool, 0x00, sizeof(struct bt_keys) * CONFIG_BT_MAX_PAIRED);
 }
 
+void fill_key_pool_by_id_addr(const struct id_addr_pair src[], int size, struct bt_keys *refs[])
+{
+	uint8_t id;
+	bt_addr_le_t *addr;
+	struct id_addr_pair const *params_vector;
+
+	zassert_true(check_key_pool_is_empty(),
+		     "List isn't empty, make sure to run this test just after a fresh start");
+
+	for (size_t it = 0; it < size; it++) {
+		params_vector = &src[it];
+		id = params_vector->id;
+		addr = params_vector->addr;
+		refs[it] = bt_keys_get_addr(id, addr);
+		zassert_true(refs[it] != NULL, "bt_keys_get_addr() returned a non-valid reference");
```

Comment:
I would suggest `Failed to add key to the keys pool`.
The existing message is technically correct but won't help much when debugging (and the debugger has forgotten the weird behavior of `bt_keys_get_addr`).

Reply:
From your previous comment here, I thought that you aren't in favor of using any `zassert_` checks outside the test suite setup functions or test cases.

Reply:
I'm reviewing commit by commit so I haven't made it to the end yet if you have changed this further. 
I think the assert is fine, it's just the assert message I was being nitpicky about.